cmake/public/utils.cmake

```diff
@@ -101,10 +101,8 @@ macro(caffe2_interface_library SRC DST)
     $<TARGET_PROPERTY:${SRC},INTERFACE_COMPILE_DEFINITIONS>
     INTERFACE_COMPILE_OPTIONS
     $<TARGET_PROPERTY:${SRC},INTERFACE_COMPILE_OPTIONS>
-    INTERFACE_INCLUDE_DIRECTORIES
-    $<TARGET_PROPERTY:${SRC},INTERFACE_INCLUDE_DIRECTORIES>
     INTERFACE_SYSTEM_INCLUDE_DIRECTORIES
-    $<TARGET_PROPERTY:${SRC},INTERFACE_SYSTEM_INCLUDE_DIRECTORIES>)
+    "$<TARGET_PROPERTY:${SRC},INTERFACE_SYSTEM_INCLUDE_DIRECTORIES>;$<TARGET_PROPERTY:${SRC},INTERFACE_INCLUDE_DIRECTORIES>")
```

Comment:
Note that SYSTEM headers are not supported for MSVC in the current version of CMake.

Reply:
I assume you mean that it'll still generate warnings on msvc. In that case, I am at a bit of a loss on what to do. I still think that propagating the compile options is not ideal, but I'm not sure to what degree generating warnings on windows is acceptable.